src/server.cpp

```diff
@@ -72,56 +72,197 @@ int init() {
 	createServerPipe(serverPath);
 	return 0;
 }
+/*
+ * @brief checks sign before string/integer
+ *
+ * returns:
+ * 		0: no sign (==)
+ * 		1: >
+ * 		2: >=
+ * 		3: <
+ * 		4: <=
+ * 		5: *
+ */
+int checkIfSign(string pattern) {
+	if (pattern.size() > 1) {
+		if (pattern[0] == '>' && !(pattern[1] == '='))
+			return 1;
+		else if (pattern[0] == '>' && pattern[1] == '=')
+			return 2;
+		else if (pattern[0] == '<' && !(pattern[1] == '='))
+			return 3;
+		else if (pattern[0] == '<' && pattern[1] == '=')
+			return 4;
+	}
+	if (pattern.size() == 1 && pattern[0] == '*')
+		return 5;
+	return 0; //only 1 char, cannot be sign
+}
+
+/*
+ * @brief compares string lexically
+ *
+ * returns:
+ * 		-1: str1 < str2
+ * 		 0: str1 == str2
+ * 		 1: str1 > str2
+ */
+int compareLex(string str1, string str2) {
+	for (unsigned i = 0; i < min(str1.size(), str2.size()); ++i) {
+		if (str1[i] == str2[i] || str2[i] == '*')
+			continue;
+		else if (str1[i] < str2[i])
+			return -1;
+		else if (str1[i] > str2[i])
+			return 1;
+	}
+	if (str1.size() < str2.size())
+		return -1;
+	else if (str1.size() > str2.size())
+		return 1;
+	else //same size
+		return 0;
+}
+
+/*
+ * @brief checks if element matches pattern
+ */
+bool checkPattern(Elem el1, Elem el2) {
```

Comment:
It would be useful to note (somewhere) that the second el2 is pattern (I mean in the context of line 165). e.g. el1 and el2 names could be more meaningfull

Reply:
good point